src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java

```diff
@@ -166,6 +171,36 @@ private void handleClientList(RoutingContext rc) {
         }
     }
 
+    private void handleClientListBySite(RoutingContext rc) {
+        try {
+            final Site site = RequestUtil.getSiteFromUrl(rc, "siteId", this.siteProvider);
+            if (site == null) {
+                return;
+            }
+
+            JsonArray ja = new JsonArray();
+            List<LegacyClientKey> collection = this.clientKeyProvider.getAll().stream().filter(legacyClientKey -> legacyClientKey.getSiteId() == site.getId()).collect(Collectors.toList());
+            for (LegacyClientKey c : collection) {
+                JsonObject jo = new JsonObject();
+                ja.add(jo);
+
+                jo.put("key_id", c.getKeyId());
```

Comment:
What do you think about giving this a similar refactoring to what you did for the [SiteService](https://github.com/IABTechLab/uid2-admin/pull/184/files#diff-f6fd75edc9be9de7cd9f625255750e10d7b1d2df3093eafe7fc042bbe4dc8b8f)? That way `handleClientList` and `handleClientListBySite` don't have so much duplicated code.

Reply:
I think either way is valid. I left it this way because I felt it would encourage removing data from the object more if there's data we don't want to pass to the portal and we can't quite as accidentally pass new data.

Reply:
alrighty - we can always refactor later if we change our minds.